libraries/src/CMS/Application/ApiApplication.php

```diff
@@ -135,11 +150,32 @@ public function getTemplate($params = false)
 	protected function route()
 	{
 		$uri = \JUri::getInstance();
-		$router = static::getRouter();
+		$router = $this->getApiRouter();
 
 		// Trigger the onBeforeApiRoute event.
 		PluginHelper::importPlugin('webservices');
 		$this->triggerEvent('onBeforeApiRoute', &$router);
+
+		$route = $router->parseRoute($uri::current(), $this->input->getMethod());
+
+		$this->input->set('option', $route['vars']['component']);
+		$this->input->set('controller', $route['controller']);
+		$this->input->set('task', $route['task']);
+
+		foreach ($route['vars'] as $key => $value)
+		{
+			if ($key !== 'component')
+			{
+				if ($this->input->getMethod() === 'POST')
```

Comment:
I'm not sure this is right. Path variables aren't really part of `$_POST` (which is what `$this->input->post` maps to)